update how-to guide for csharp (#188)
Azure/typespec-azure: docs/howtos/DataPlane Generation - DPG/03convenient.mdx

<TabItem value="csharp" label="CSharp" >

```csharp
// Model class is internal
namespace PetStoreNamespace.Models
{
  internal partial class OutputModel
  {
    public string Name { get; }
  }
}
// Client method is internal
namespace PetStoreNamespace
{
  public partial class PetStoreNamespaceClient
  {
    //protocol method
    internal virtual async Task<Response> GetModelAsync(string name, RequestContext context)
    internal virtual Response GetModel(string name, RequestContext context)
    //convenience method
    internal virtual async Task<Response<OutputModel>> GetModelAsync(string name, CancellationToken cancellationToken = default)

>     internal virtual Response<OutputModel> GetModel(string name, CancellationToken cancellationToken = default)

Reviewer: we should add curly braces for methods.

<TabItem value="csharp" label="CSharp" >

```csharp

> // If a model is input-only, there will be one public constructor with all required properties as parameter.

Reviewer: ```suggestion
// If a model is input-only, it has one public constructor with all required properties as parameters.
```

> // And the required properties without setter. Optional properties will have both getter and setter

Reviewer: ```suggestion
// The required properties only have getter. Optional properties have both getter and setter.
```

Reviewer: Should we also mention about the setter of collections?

Author: add part for collection properties

namespace Azure.AI.OpenAI.Models
{
  public partial class AzureCognitiveSearchIndexFieldMappingOptions
  {
    public AzureCognitiveSearchIndexFieldMappingOptions()

    public string TitleFieldName { get; set; }
    public string UrlFieldName { get; set; }
    public string FilepathFieldName { get; set; }
    public IList<string> ContentFieldNames { get; }
    public string ContentFieldSeparator { get; set; }
    public IList<string> VectorFieldNames { get; }
    public IList<string> ImageVectorFieldNames { get; }
  }
}

> // If a model is output-only, there is no public constructor, and all properties only have getter, no setter.

Reviewer: ```suggestion
// If a model is output-only, it has no public constructor, and all properties only have getter, no setter.
```

// If a model is roundtrip (input + output), there will be one public constructor with all required properties as parameter.

> // And all properties will have both getter and setter

Reviewer: ```suggestion
// All properties have both getter and setter
```

We should mention collection properties, because despite here we said "all properties have both getter and setter", the code below, `Data` property does not have a setter. Questions will pop up.

Author: added
